orttraining/orttraining/core/graph/gradient_builder.cc

```diff
@@ -450,8 +450,15 @@ IMPLEMENT_GRADIENT_BUILDER(GetGemmGradient) {
                       {IA("dC_reduced"), SCALE},
                       {dC}));
         } else {
-          result.push_back(
-              NodeDef("Identity", {IA("dC_reduced")}, {dC}));
+          // skip Identity to avoid memcpy between ReduceSum and graph gradient output
+          NodeDef node_def = result.back();
+          if (node_def.output_args[0] == IA("dC_reduced") && node_def.op_type == "ReduceSum") {
+            node_def.output_args[0] = dC;
+            result.back() = node_def;
+          } else {
+            result.push_back(
```

Comment:
There are other code paths within GetGemmGradient that add the Identity node. Can we remove them as well?

Reply:
I'll close this PR and investigate EliminateIdentity instead.
Thanks for the review.

---
In reply to: [610763655](https://github.com/microsoft/onnxruntime/pull/7301#discussion_r610763655) [](ancestors = 610763655)